builtin/logical/mysql/secret_creds.go

```diff
@@ -82,22 +82,33 @@ func (b *backend) secretCredsRevoke(
 	}
 
 	// Get the role name
+	// we may not always have role data in the secret InternalData
+	// so don't exit if the roleNameRaw fails. Instead it is set
+	// as an empty string
+	var roleName string
 	roleNameRaw, ok := req.Secret.InternalData["role"]
-	roleName, ok := roleNameRaw.(string)
+	if !ok {
+		roleName = ""
+	} else {
+		roleName, ok = roleNameRaw.(string)
```

Comment:
If you're not going to check `ok` here you shouldn't assign to it -- e.g. `roleName, _ = ...`